src/data/matrix/basic.lean

```diff
@@ -120,6 +116,40 @@ lemma map_smul [has_scalar R α] [has_scalar R β] (f : α →[R] β) (r : R)
   (M : matrix m n α) : (r • M).map f = r • (M.map f) :=
 by { ext, simp, }
 
+/-- The transpose of a matrix. -/
+def transpose (M : matrix m n α) : matrix n m α
+| x y := M y x
+
+localized "postfix `ᵀ`:1500 := matrix.transpose" in matrix
+
+/-- The conjugate transpose of a matrix defined in term of `star`. -/
+def conj_transpose [has_star α] (M : matrix m n α) : matrix n m α :=
+M.transpose.map star
+
+localized "postfix `ᴴ`:1500 := matrix.conj_transpose" in matrix
+
+/-- Proposition `matrix.is_sym`. `A.is_sym` means `Aᵀ = A`. -/
+protected def is_sym (A : matrix m m α) : Prop := Aᵀ = A
+
+/-- Proposition `matrix.is_skewsym`. `A.is_skewsym` means `-Aᵀ = A` if `[has_neg α]` -/
+protected def is_skewsym [has_neg α] (A : matrix m m α) : Prop := -Aᵀ = A
+
+/-- Proposition `matrix.is_Hermitian`. `A.is_Hermitian` means `Aᴴ = A` if `[has_star α]` -/
+protected def is_Hermitian [has_star α] (A : matrix m m α) : Prop := Aᴴ = A
+
+/-- Proposition `matrix.is_perm`.
+    `P.is_perm` means `P` is a permutation matrix.
+    A new file is probably required for systematically investigating permutation matrices. -/
+protected def is_perm [decidable_eq m] [has_zero α] [has_one α] (P : matrix m m α): Prop :=
+∃ σ : equiv.perm m, P = equiv.perm.to_matrix σ
+
+/-- Proposition `matrix.is_perfect_shuffle`.
+    `P.is_perfect_shuffle` means `P` is a perfect shuffle matrix.
+    A new file is probably required for systematically investigating perfect shuffle matrices. -/
+protected def is_perfect_shuffle
+[decidable_eq m] [has_zero α] [has_one α] (P : matrix m m α) : Prop :=
+∃ σ : equiv.perm m, (P = equiv.perm.to_matrix σ ∧ ∀ i : m, σ i ≠ i)
```

Comment:
These should move to `matrix/pequiv.lean` too - although other reviewers may still decide we don't want them at all.